src/Rules/IRuleMetadataProvider.cs

```diff
@@ -39,34 +42,55 @@ public interface IRuleMetadataProvider
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class RuleMetadataProvider : IRuleMetadataProvider
     {
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        public RuleMetadataProvider(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
         public IRuleHelp GetRuleHelp(Language language, string ruleKey)
         {
-            string description = language?.ServerLanguage?.Key == null 
+            string description = language?.ServerLanguage?.Key == null
                 ? Resources.Rules_DescriptionForMissingRule
                 : GetDescription(language, ruleKey);
 
             return new RuleHelp(language, ruleKey, description);
         }
 
         private string GetDescription(Language language, string ruleKey)
+        {
+            var ruleInfo = LoadRuleInfo(language, ruleKey);
+            return ruleInfo?.Description ?? Resources.Rules_DescriptionForMissingRule;
```

Comment:
I avoided changing the `IRuleMetadataProvider` interface to limit the scope of the changes in this PR.

The class now uses the new `IRuleInfo` class internally to load the embedded json, but just uses it to create and return an instance of `IRuleHelp`.